src/theory/quantifiers/fun_def_process.cpp

```diff
@@ -29,7 +29,7 @@ using namespace CVC4::theory::quantifiers;
 using namespace CVC4::kind;
 
 
-void FunDefFmf::simplify( std::vector< Node >& assertions, bool doRewrite ) {
+void FunDefFmf::simplify( std::vector< Node >& assertions ) {
```

Comment:
I think this would be expensive, since this would require a copy the vector of input assertions.  

For some context, we call simplify as a preprocessing pass on the assertion vector:
https://github.com/CVC4/CVC4/blob/master/src/smt/smt_engine.cpp#L4117

Regardless, this is something that should be refactored once we standardize the preprocessing passes in SmtEnginePrivate::processAssertions, where then the FunDefFmf should conform to whatever that infrastructure expects. (I guess this is being worked on in #1053).

Reply:
SG then.